scripts/bench_python_cpp_algos.py

```diff
@@ -353,73 +365,67 @@ def load_graph(graph_path, edge_properties=None):
 
     # Load our graph
     input = next(item for item in inputs if item["name"] == args.graph)
-    data = create_empty_statistics()
-
-    if args.application in ["bfs", "sssp", "bc", "jaccard"]:
-        graph_path = f"{args.input_dir}/{input['name']}"
-        if not os.path.exists(graph_path):
-            print(f"Graph doesn't exist: {graph_path}")
-
-        graph = load_graph(graph_path)
-
-        if args.application == "bfs":
-            data = run_routine(run_bfs, data, args.trials, (graph, input, args.source_nodes))
-
-        if args.application == "sssp":
-            data = run_routine(run_sssp, data, args.trials, (graph, input, args.source_nodes))
-
-        if args.application == "jaccard":
-            data = run_routine(run_jaccard, data, args.trials, (graph, input))
-
-        if args.application == "bc":
-            data = run_routine(run_bc, data, args.trials, (graph, input, args.source_nodes, 4))
+    data = create_empty_statistics(args)
+
+    routine_name_args_mappings = {
+        "bfs": [bfs, (args.source_nodes)],
+        "sssp": (sssp, (args.source_nodes)),
+        "jaccard": (jaccard, ()),
+        "bc": (bc, (args.source_nodes, args.num_sources)),
+        "tc": (tc, ()),
+        "cc": (cc, ()),
+        "kcore": (kcore, ()),
+        "louvain": (louvain, ()),
+        "pagerank": (pagerank, ()),
```

Comment:
It might be worth defining a namedtuple for this. That way things are a bit clearer everywhere this data is used. The [0] and [1] below are pretty opaque.